### PR TITLE
DOMA-4058 fixed history changes after auto reopen ticket

### PR DESCRIPTION
**Pull request by Alllex202**
Before:
Снимок экрана 2022-08-31 в 17 57 20

After:
Снимок экрана 2022-08-31 в 17 58 03

### CODE REVIEW

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 1]
```diff
@@ -316,11 +317,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
...
+                message: formatAutoReopenDiffMessage(field, ticketChange),
+            }))
+    }
+
     // Omit what was not changed
     let changedFields = fields.filter(([field]) => (
         ticketChange[`${field}From`] !== null || ticketChange[`${field}To`] !== null
     ))
 
+    if (ticketChange.sender.fingerprint === 'auto-reopen') {
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:398]: ok, on line 264 it is also used.
this is the most efficient and simple method.
need to do additional checks for changes in the expected fields?

[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:398]: I don't have enough business context to advise. But, you need to find a better way. This field is not for business logic

**Comment by Alllex202** [comment]
Also:
 - refactored logic without use sender field. 
 - fixed color status in auto-close ticket change message
     - Before:
Снимок экрана 2022-09-01 в 17 20 20
     - After: 
Снимок экрана 2022-09-01 в 17 20 03

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 2]
```diff
@@ -316,11 +333,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
...
+
+            return (
+                <FormattedMessage
+                    id='pages.condo.ticket.TicketChanges.autoReopenTicket.deferredUntil'
+                    values={{
+                        deferredUntil: valueDeferredUntilFrom,
+                    }}
+                />
+            )
+        } else if (field === 'statusDisplayName') {
```
Comment (dkoviazin) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:349]: Too many else if for one field conditions
You can use object with keys for field value 
Or switch ... case

**Review by dkoviazin — APPROVED** [review]

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 3]
```diff
@@ -316,11 +333,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
             )
         }
     }
+
+    const formatAutoReopenDiffMessage = (field, ticketChange) => {
+        if (field === 'deferredUntil') {
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:338]: Use object/dict like elseIf pattern

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 4]
```diff
@@ -260,17 +270,24 @@ const useChangedFieldMessagesOf = (ticketChange) => {
...
         const valueFrom = ticketChange[`${field}From`]
         const valueTo = ticketChange[`${field}To`]
         const isValueFromNotEmpty = !isNil(valueFrom)
         const isValueToNotEmpty = !isNil(valueTo)
         const formattedValueFrom = formatField(field, valueFrom, TicketChangeFieldMessageType.From)
         const formattedValueTo = formatField(field, valueTo, TicketChangeFieldMessageType.To)
 
+        if (isAutoCloseTicketChanges(ticketChange)) {
+            return (
+                <FormattedMessage
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:282]: we use another pattern to formatted messages. Don't use FormattedMessage tag

Comment (Alllex202) [reply apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx]: refactored it

**Pull request merged**